src/dfx/src/lib/models/canister.rs

```diff
@@ -432,12 +499,18 @@ fn build_canister_js(canister_id: &CanisterId, canister_info: &CanisterInfo) ->
                 &canister_info.get_name().to_uppercase(),
             );
 
-        match decode_path_to_str(&file.path()?)? {
+        match decode_path_to_str(&file.path()?).context("Failed to decode path to str.")? {
```

Comment:
I think this context is now reported by the `decode_path_to_str` itself
```suggestion
        match decode_path_to_str(&file.path()?)? {
```